pkg/xt/xt.go

```diff
@@ -31,13 +32,21 @@ func Extract(job *Job) {
 
 			start := time.Now()
 
-			size, files, _, err := xtractr.ExtractFile(&xtractr.XFile{
+			xFile := &xtractr.XFile{
 				FilePath:  fileName,            // Path to archive being extracted.
 				OutputDir: job.Output,          // Folder to extract archive into.
 				FileMode:  job.FileMode.Mode(), // Write files with this mode.
 				DirMode:   job.DirMode.Mode(),  // Write folders with this mode.
 				Passwords: job.Passwords,       // (RAR/7zip) Archive password(s).
-			})
+			}
+
```

Comment:
We should add some code here that does something like:
- remove input path prefix from fileName.
- append fileName Dir to job.Output.
- extract file into job.Output/fileFolder

To keep your current implementation, one would pass the same path to `-o` that was provided to search.

Do you want to tackle this? if not, I'll merge your contribution and do it myself.